### PR TITLE
Add contribution guide for third-party library and fix pre-commit

### PR DESCRIPTION
**Pull request by ejguan**
- Add contribution guide for third-party library
- Change pre-commit to use lower `node` version

### CODE REVIEW

File: third_party/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Third-Party Libraries
+
+The `third_party` directory contains all of (optional) dependency libraries used by `torchdata`.
```
Comment (NivekT) [inline third_party/CONTRIBUTING.md:3]:
```suggestion
The `third_party` directory contains all of the (optional) dependency libraries used by `torchdata`.
```

File: third_party/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+## Integration Requirement
+
+### Soft Dependency vs Hard Dependency
+
+`TorchData` as a data-processing libraries provides a bunch of `DataPipes` that are integrated with
+different third-party libraries to handle specific use cases. For example, `datasets` is imported
+to load dataset from `HuggingFace`, and `fsspec` is imported to provide a unified API to access and
+load from local or remote file systems. Those dependencies should only be verified of their availability
+until they are used/referenced by users during `DataPipe` construction. You can find examples from
```
Comment (NivekT) [inline third_party/CONTRIBUTING.md:19]:
```suggestion
load from local or remote file systems. Those dependencies are optional and should only be verified of their availability
when they are used/referenced by users during `DataPipe` construction. You can find examples from
```

File: third_party/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+[here](https://github.com/pytorch/data/blob/bb78231e5f87620385cb2f91cda87e7f9414eb4a/torchdata/datapipes/iter/load/huggingface.py#L57-L62)
+and [here](https://github.com/pytorch/data/blob/d19858202df7e8b75765074259e6023f539cbf3f/torchdata/datapipes/iter/load/fsspec.py#L59).
```
Comment (NivekT) [inline third_party/CONTRIBUTING.md:21]:
```suggestion
and [here](https://github.com/pytorch/data/blob/d19858202df7e8b75765074259e6023f539cbf3f/torchdata/datapipes/iter/load/fsspec.py#L59).
They are contrasted with Core dependencies that are must be installed along with `torchdata`.
```

**Review by NivekT — APPROVED** [review]
LGTM! A few nit changes, feel free to ignore if you do not find them helpful.

**Review by VitalyFedyunin — APPROVED** [review]

File: third_party/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+## Integration Requirement
+
+### Soft Dependency vs Hard Dependency
+
+`TorchData` as a data-processing libraries provides a bunch of `DataPipes` that are integrated with
+different third-party libraries to handle specific use cases. For example, `datasets` is imported
+to load dataset from `HuggingFace`, and `fsspec` is imported to provide a unified API to access and
+load from local or remote file systems. Those dependencies are optional and should only be verified
+of their availability when they are used/referenced by users during `DataPipe` construction.You can
```
Comment (NivekT) [inline third_party/CONTRIBUTING.md:19]:
```suggestion
of their availability when they are used/referenced by users during `DataPipe` construction. You can
```

**Review by NivekT — APPROVED** [review]
nit

**Pull request closed without merging**